### DISCUSSION TITLE
code is imploding on borderRadii

### DISCUSSION THREAD

**Discussion by gjblajian** [csandman/chakra-react-select#94] [Q&A] [answered]
https://github.com/csandman/chakra-react-select#size--options-sm-md-lg--default-md
@csandman we are using the select component and it is rendering fine at first but the minute we click on the component the underlying menuList is blowing up. It seems to be indicating that the theme is not loaded. The size attribute is set to sm but it also blows up in the same location if the attribute is not set (defaulting to md). 

So my question is, do we need to include some other library to use the useTheme().radii? 

```
menu.tsx:85 Uncaught TypeError: Cannot read properties of undefined (reading 'md')
    at MenuList (menu.tsx:85:1)
    at renderWithHooks (react-dom.development.js:14985:1)
    at mountIndeterminateComponent (react-dom.development.js:17811:1)
    at beginWork (react-dom.development.js:19049:1)
    at HTMLUnknownElement.callCallback (react-dom.development.js:3945:1)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:3994:1)
    at invokeGuardedCallback (react-dom.development.js:4056:1)
    at beginWork$1 (react-dom.development.js:23964:1)
    at performUnitOfWork (react-dom.development.js:22776:1)
    at workLoopSync (react-dom.development.js:22707:1)
```

```
  const borderRadii: SizeProps = useTheme().radii;

  const initialStyles: SystemStyleObject = {
    ...list,
    maxHeight: `${maxHeight}px`,
    overflowY: "auto",
    borderRadius: borderRadii[size as Size], // blowing up on this line
  };
```

**Comment by csandman** [comment]
You do need [Chakra UI](https://chakra-ui.com/guides/first-steps) installed to use this package, and the [`ChakraProvider`](https://chakra-ui.com/guides/getting-started/cra-guide#2-provider-setup) component needs to be above `chakra-react-seelct` in your component tree.  So if you don't have that set up, then that would be your issue.

If you do have that set up already, then something else must be wrong and I'd need more info to diagnose.  If that doesn't fix it, can you post a full code snippet of how you implemented the `Select` component?

**Reply by jaredmpeterson** [reply]
@csandman our problem was with `"react-select": "5.3.1"` reverting to `5.3.0` fixed our problem.
CC: @gjblajian

**Reply by csandman** [reply]
@jaredmpeterson was the issue in combination with using react 18?

**Reply by jaredmpeterson** [reply]
@csandman we haven't upgraded to v18 yet. We upgraded to `5.3.1` on 2 May 22, and didn't notice the issue until yesterday. We have a few implementations of chakra-react-select (mostly to handle multi-select), we use Chakra's select for most cases. I can share snippets if needed.

**Reply by csandman** [reply]
If you have more info/snippets, I'm curious.  Were you running into the same issue as @gjblajian? I just bumped the dependency for `react-select` to `v5.3.1` in version `3.3.2` so if that's going to cause issues I'd definitely like to know about it.

From what I can tell, the only difference in `3.3.2` is updating the minimum version of `@emotion/react` to `11.8.1`, so perhaps it had something to do with conflicting versions of `@emotion/react` in your project vs. `react-select`?

**Reply by jaredmpeterson** [reply]
@csandman, my apologies @gjblajian and I are on the same team.

**Reply by csandman** [reply]
Ah haha, ok that wasn't clear.  I'm still curious what caused `react-select@5.3.1` to break your app...

**Reply by jaredmpeterson** [reply]
we have `@emotion/react@11.9.0` installed as a dependency, so I didn't think this would be a problem for us

[react-select@5.3.1 ](https://github.com/JedWatson/react-select/releases/tag/react-select%405.3.1)[Latest](https://github.com/JedWatson/react-select/releases/latest)
Patch Changes

[03bf7351](https://github.com/JedWatson/react-select/commit/03bf735127fec6e47de8ae45e7cdc0a39c8b638b) https://github.com/JedWatson/react-select/pull/5164 Thanks [@Methuselah96](https://github.com/Methuselah96)! - Bump @emotion/react to ^11.8.1 to avoid useInsertionEffect bug

**Reply by csandman** [reply]
Yeah so if the only change in version `5.3.1` is the bump from `@emotion/react` to `11.8.1`, and your project already has a higher version than that (meaning they should be de-duped on install), I can't see how this would cause you any issues.

Can you share what version of `npm` or `yarn` you're using?

**Reply by jaredmpeterson** [reply]
yarn --version = 3.2

**Reply by csandman** [reply]
Oh interesting, I don't have much experience with yarn so it's possible that has something to do with it.  In order to make this package work with yarn@3, did you have to install all of the individual chakra peer dependencies (`@chakra-ui/system`, `@chakra-ui/form-control` etc.) in addition to `@chakra-ui/react`?

**Reply by jaredmpeterson** [reply]
No, yarn@3 has actually been really nice. We were using lerna + yarn workspaces (yarn@1) for monorepo management. But migrated to just yarn@3. This has been the only instance of a package giving us issues in 2+ months.

**Reply by jaredmpeterson** [reply]
@csandman I updated `chakra-react-select@3.3.3` and removed our dependency for `react-select` and we are good to go.

**Reply by csandman** [reply]
Glad to hear it! I didn't realize you had a separate dep on that, do you think that was causing the issue?

**Reply by jaredmpeterson** [reply]
probably. It was installed and used months ago before we migrated to `chakra-react-select`.

**Reply by csandman** [reply]
Well thank you for responding with how you got it working again, this might help others in the future!

**Reply by jaredmpeterson** [reply]
We can mark it as answered. 😄

**Reply by csandman** [reply]
> @csandman I updated `chakra-react-select@3.3.3` and removed our dependency for `react-select` and we are good to go.

If you'd like to repost your comment from above as a new answer and mark it as the correct answer that could make it easier for others to find it!

**Comment by csandman** [comment, marked as answer]
> @csandman I updated chakra-react-select@3.3.3 and removed our dependency for react-select and we are good to go.
